### PR TITLE
Opt in documentation

### LINKED ISSUE #20: Add info regarding opt-in attributes
The docs should include a warning/info for attributes that require opt in. The user should be also educated on how to opt-in.

This includes:
- `item_preview_hero_path`
- `display_asset_path`
- `definition_path`
- `path`
- `gameplay_tags`
- `shop_history`

### PR DESCRIPTION
**Pull request by trevorflahardy**
## Summary

This PR updates documentation to include information regarding the newly added `ResponseFlags` class in the library.

- Updates the migration guide to include the `response_flags` parameter in the client.
- Creates a `Response Flags` section in the documentation describing what response flags are and how to use them.
- Creates a `.. opt-in` sphinx directive that marks an attribute as needing to be marked-in.
- Updates all relevant attributes with said sphinx directive. 

image
image

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: fortnite_api/shop.py [hunk 1]
```diff
@@ -439,6 +439,8 @@ class ShopEntry(ReconstructAble[Dict[str, Any], HTTPClientT]):
         The offer ID of this entry.
     display_asset_path: Optional[:class:`str`]
         The display asset path of this entry.
+
+        .. opt-in:: INCLUDE_PATHS
```
Comment (Luc1412) [inline fortnite_api/shop.py]: This field isn't affected by opt-in params.

File: fortnite_api/playlist.py [hunk 2]
```diff
@@ -111,8 +111,12 @@ class Playlist(Hashable, ReconstructAble[Dict[str, Any], HTTPClientT]):
         The images associated with the playlist.
     gameplay_tags: List[:class:`str`]
         The gameplay tags for the playlist.
+
+        .. opt-in:: INCLUDE_GAMEPLAY_TAGS
     path: :class:`str`
         The path of the playlist.
+
+        .. opt-in:: INCLUDE_PATHS
```
Comment (Luc1412) [inline fortnite_api/playlist.py]: Both fields are not affected by opt-in params.

File: docs/response_flags.rst
Comment (Luc1412) [inline docs/response_flags.rst:1]: I think we should emphasize the user to only use the required opt-in parameters they actually need. And how the user benefits from it. (e.g. perfomance, bandwith)

File: fortnite_api/cosmetics/br.py [hunk 3]
```diff
@@ -241,14 +243,24 @@ class CosmeticBr(Cosmetic[Dict[str, Any], HTTPClientT]):
         The dynamic pak ID of the cosmetic, if available.
     item_preview_hero_path: Optional[:class:`str`]
         The item preview hero path of the cosmetic, if available.
+
+        .. opt-in:: INCLUDE_PATHS
     display_asset_path: Optional[:class:`str`]
         The display asset path of the cosmetic, if available.
+
+        .. opt-in:: INCLUDE_PATHS
```
Comment (Luc1412) [inline fortnite_api/cosmetics/br.py:251]: This attribute seems to not present anymore. I need to check back with Officer about it's status.

**Review by Luc1412 — CHANGES_REQUESTED** [review]
Looks good, but directive is applied to attributes that are not part of the opt-in.

File: fortnite_api/cosmetics/br.py (hunk 3 above)
Comment (Luc1412) [reply fortnite_api/cosmetics/br.py:251]: Officer recommends to keep this for now

File: docs/response_flags.rst
Comment (trevorflahardy) [reply docs/response_flags.rst:1]: Added in 5270136 in an `important` directive that explains why you should not enable all flags by default, an example of it in action, and how you should develop with response flags in mind.

**Review by Luc1412 — APPROVED** [review]
lgtm

**Pull request merged**